lib/srv/server/ec2_watcher.go

```diff
@@ -146,26 +188,98 @@ func newEC2InstanceFetcher(cfg ec2FetcherConfig) *ec2InstanceFetcher {
 	return &fetcherConfig
 }
 
+func (f *ec2InstanceFetcher) GetMatchingInstances(nodes []types.Server, rotation bool) ([]Instances, error) {
+	insts := EC2Instances{
+		Region:       f.Region,
+		DocumentName: f.DocumentName,
+		Parameters:   f.Parameters,
+		Rotation:     rotation,
+	}
+	for _, node := range nodes {
+		if node.GetSubKind() != types.SubKindOpenSSHNode {
+			continue
+		}
+		region, ok := node.GetLabel(types.AWSInstanceRegion)
+		if !ok || region != f.Region {
+			continue
+		}
+		instID, ok := node.GetLabel(types.AWSInstanceIDLabel)
+		if !ok {
+			continue
+		}
+		accountID, ok := node.GetLabel(types.AWSAccountIDLabel)
+		if !ok {
+			continue
+		}
+
+		_, ok = f.foundInstances[foundInstancesKey{accountID: accountID, instanceID: instID}]
+		if !ok {
+			continue
+		}
+		if insts.AccountID != "" {
```

Comment:
Should this be `==` rather than `!=`? Otherwise seems strange, you're overwriting it if it's not empty?